### ISSUE TITLE
debug port-forward does not work on 4.6 cluster

### ISSUE THREAD

**Issue by prietyc123** [openshift/odo#4036]
/kind bug

## What versions of software are you using?

**Operating System:**
All supported

**Output of `odo version`:**
master

## How did you run odo exactly?

```
$ odo create --s2i nodejs
Validation
Warning: nodejs is not fully supported by odo, and it is not guaranteed to work
 ✓  Validating component [817ms]

Please use `odo push` command to create the component with source deployed

$ odo push
Validation
 ✓  Checking component [2s]

Configuration changes
 ✓  Initializing component
 ✓  Creating component [3s]

Applying URL changes
 ✓  URLs are synced with the cluster, no changes are required.

Pushing to component nodejs-testdev-yjzp of type local
 ✓  Checking files for pushing [1ms]
 ✓  Waiting for component to start [30s]
 ✓  Syncing files to the component [3s]
 ✓  Building component [16s]
 ✓  Changes successfully pushed to component

$ odo debug port-forward --local-port 54753 --context .
Started port forwarding at ports - 54753:5858
E0924 19:55:58.468770   88386 portforward.go:400] an error occurred forwarding 54753 -> 5858: error forwarding port 5858 to pod 3951a99486aeaa4347a76b6a665f1c8fb60c5a12160e715a6fbd71569867dee0, uid : port forward into network namespace "/var/run/netns/6fc98c88-77b6-4339-ad27-79b9191d907b": dialing 5858: dial tcp4 127.0.0.1:5858: connect: connection refused
E0924 19:55:58.470007   88386 portforward.go:400] an error occurred forwarding 54753 -> 5858: error forwarding port 5858 to pod 3951a99486aeaa4347a76b6a665f1c8fb60c5a12160e715a6fbd71569867dee0, uid : port forward into network namespace "/var/run/netns/6fc98c88-77b6-4339-ad27-79b9191d907b": dialing 5858: dial tcp4 127.0.0.1:5858: connect: connection refused
E0924 19:55:59.283965   88386 portforward.go:400] an error occurred forwarding 54753 -> 5858: error forwarding port 5858 to pod 3951a99486aeaa4347a76b6a665f1c8fb60c5a12160e715a6fbd71569867dee0, uid : port forward into network namespace "/var/run/netns/6fc98c88-77b6-4339-ad27-79b9191d907b": dialing 5858: dial tcp4 127.0.0.1:5858: connect: connection refused
E0924 19:56:00.719741   88386 portforward.go:400] an error occurred forwarding 54753 -> 5858: error forwarding port 5858 to pod 3951a99486aeaa4347a76b6a665f1c8fb60c5a12160e715a6fbd71569867dee0, uid : port forward into network namespace "/var/run/netns/6fc98c88-77b6-4339-ad27-79b9191d907b": dialing 5858: dial tcp4 127.0.0.1:5858: connect: connection refused
E0924 19:56:30.312872   88386 portforward.go:362] error creating forwarding stream for port 54753 -> 5858: Timeout occurred
E0924 19:56:32.191587   88386 portforward.go:340] error creating error stream for port 54753 -> 5858: Timeout occurred
E0924 19:56:32.874889   88386 portforward.go:400] an error occurred forwarding 54753 -> 5858: error forwarding port 5858 to pod 3951a99486aeaa4347a76b6a665f1c8fb60c5a12160e715a6fbd71569867dee0, uid : port forward into network namespace "/var/run/netns/6fc98c88-77b6-4339-ad27-79b9191d907b": dialing 5858: dial tcp4 127.0.0.1:5858: connect: connection refused
```

## Actual behavior

When I browse the `http://localhost:54753`, it throws connection refused. The same behaviour I have observed through the test script as well.

## Expected behavior

`odo debug port-forward` should work.

**Comment by prietyc123** [comment]
I set the priority as high because its blocking 4.6 cluster to enable in periodic job against prow.

**Comment by kadel** [comment]
This issue is related to https://github.com/openshift/odo/issues/4016
Components are by default executed in debug mode only for supported builder images. So once https://github.com/openshift/odo/issues/4016 is resoled by adding new builder images used by 4.6 as a supported image this will get automatically resolved.

**Comment by dgolovin** [comment]
@kadel I am hitting the same problem for both s2i and devfile nodejs components.
Using crc v1.15.0 with OpenShift v4.5.7

**Comment by prietyc123** [comment]
> @kadel I am hitting the same problem for devfile nodejs components.
> Using crc v1.15.0 with OpenShift v4.5.7

@dgolovin Could you please let us know the imagestream details of nodejs on the cluster you are using.

**Comment by rnapoles-rh** [comment]
This is the output I get on 4.5.7 (from cluster-bot):

**Comment by rnapoles-rh** [comment]
**Running on CRC:**
$ crc version
CodeReady Containers version: 1.15.0+e317bed
OpenShift version: 4.5.7 (embedded in binary)

$ oc version
Client Version: 4.5.7
Kubernetes Version: v1.18.3+2cf11e2

```
oc -n openshift get is java -o yaml
apiVersion: image.openshift.io/v1
kind: ImageStream
metadata:
  annotations:
    openshift.io/display-name: Red Hat OpenJDK
    openshift.io/image.dockerRepositoryCheck: "2020-08-20T09:50:50Z"
    openshift.io/provider-display-name: Red Hat, Inc.
    samples.operator.openshift.io/version: 4.5.7
    version: 1.4.17
  creationTimestamp: "2020-08-20T09:50:47Z"
  generation: 2
  labels:
    samples.operator.openshift.io/managed: "true"
    xpaas: 1.4.17
  managedFields:
  - apiVersion: image.openshift.io/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:openshift.io/display-name: {}
          f:openshift.io/provider-display-name: {}
          f:samples.operator.openshift.io/version: {}
          f:version: {}
        f:labels:
          .: {}
          f:samples.operator.openshift.io/managed: {}
          f:xpaas: {}
      f:spec:
        f:tags:
          .: {}
          k:{"name":"8"}:
            .: {}
            f:annotations:
              .: {}
              f:description: {}
              f:iconClass: {}
              f:openshift.io/display-name: {}
              f:sampleContextDir: {}
              f:sampleRepo: {}
              f:supports: {}
              f:tags: {}
              f:version: {}
            f:from:
              .: {}
              f:kind: {}
              f:name: {}
            f:generation: {}
            f:importPolicy: {}
            f:name: {}
            f:referencePolicy:
              .: {}
              f:type: {}
          k:{"name":"11"}:
            .: {}
            f:annotations:
              .: {}
              f:description: {}
              f:iconClass: {}
              f:openshift.io/display-name: {}
              f:sampleContextDir: {}
              f:sampleRepo: {}
              f:supports: {}
              f:tags: {}
              f:version: {}
            f:from:
              .: {}
              f:kind: {}
              f:name: {}
            f:generation: {}
            f:importPolicy: {}
            f:name: {}
            f:referencePolicy:
              .: {}
              f:type: {}
          k:{"name":"latest"}:
            .: {}
            f:annotations:
              .: {}
              f:description: {}
              f:iconClass: {}
              f:openshift.io/display-name: {}
              f:sampleContextDir: {}
              f:sampleRepo: {}
              f:supports: {}
              f:tags: {}
              f:version: {}
            f:from:
              .: {}
              f:kind: {}
              f:name: {}
            f:generation: {}
            f:importPolicy: {}
            f:name: {}
            f:referencePolicy:
              .: {}
              f:type: {}
    manager: cluster-samples-operator
    operation: Update
    time: "2020-08-20T09:50:47Z"
  name: java
  namespace: openshift
  resourceVersion: "18275"
  selfLink: /apis/image.openshift.io/v1/namespaces/openshift/imagestreams/java
  uid: 01f5666a-8ef2-4d13-a1f2-8f1e53d6ef50
spec:
  lookupPolicy:
    local: false
  tags:
  - annotations:
      description: Build and run Java applications using Maven and OpenJDK 11.
      iconClass: icon-rh-openjdk
      openshift.io/display-name: Red Hat OpenJDK 11
      sampleContextDir: undertow-servlet
      sampleRepo: https://github.com/jboss-openshift/openshift-quickstarts
      supports: java:11,java
      tags: builder,java,openjdk
      version: "11"
    from:
      kind: DockerImage
      name: registry.redhat.io/openjdk/openjdk-11-rhel7:latest
    generation: 2
    importPolicy: {}
    name: "11"
    referencePolicy:
      type: Local
  - annotations:
      description: Build and run Java applications using Maven and OpenJDK 8.
      iconClass: icon-rh-openjdk
      openshift.io/display-name: Red Hat OpenJDK 8
      sampleContextDir: undertow-servlet
      sampleRepo: https://github.com/jboss-openshift/openshift-quickstarts
      supports: java:8,java
      tags: builder,java,openjdk
      version: "8"
    from:
      kind: DockerImage
      name: registry.redhat.io/redhat-openjdk-18/openjdk18-openshift:latest
    generation: 2
    importPolicy: {}
    name: "8"
    referencePolicy:
      type: Local
  - annotations:
      description: Build and run Java applications using Maven and OpenJDK 11.
      iconClass: icon-rh-openjdk
      openshift.io/display-name: Red Hat OpenJDK 11
      sampleContextDir: undertow-servlet
      sampleRepo: https://github.com/jboss-openshift/openshift-quickstarts
      supports: java:11,java
      tags: builder,java,openjdk
      version: latest
    from:
      kind: ImageStreamTag
      name: "11"
    generation: 1
    importPolicy: {}
    name: latest
    referencePolicy:
      type: Local
status:
  dockerImageRepository: image-registry.openshift-image-registry.svc:5000/openshift/java
  publicDockerImageRepository: default-route-openshift-image-registry.apps-crc.testing/openshift/java
  tags:
  - items:
    - created: "2020-08-20T09:50:50Z"
      dockerImageReference: registry.redhat.io/openjdk/openjdk-11-rhel7@sha256:9bafd6a5dd1ccf7aaa3f3bd06cfa55e15dd915cd70aaa59e5987a62b17a875d7
      generation: 2
      image: sha256:9bafd6a5dd1ccf7aaa3f3bd06cfa55e15dd915cd70aaa59e5987a62b17a875d7
    tag: "11"
  - items:
    - created: "2020-08-20T09:50:50Z"
      dockerImageReference: registry.redhat.io/redhat-openjdk-18/openjdk18-openshift@sha256:648f77558d4656107be73379219d6d2ab27a092e92a956d96737b6b0fae5000a
      generation: 2
      image: sha256:648f77558d4656107be73379219d6d2ab27a092e92a956d96737b6b0fae5000a
    tag: "8"
  - items:
    - created: "2020-08-20T09:50:50Z"
      dockerImageReference: registry.redhat.io/openjdk/openjdk-11-rhel7@sha256:9bafd6a5dd1ccf7aaa3f3bd06cfa55e15dd915cd70aaa59e5987a62b17a875d7
      generation: 2
      image: sha256:9bafd6a5dd1ccf7aaa3f3bd06cfa55e15dd915cd70aaa59e5987a62b17a875d7
    tag: latest
```

and:


```
oc -n openshift get is nodejs -o yaml
apiVersion: image.openshift.io/v1
kind: ImageStream
metadata:
  annotations:
    openshift.io/display-name: Node.js
    openshift.io/image.dockerRepositoryCheck: "2020-08-20T09:50:58Z"
    samples.operator.openshift.io/version: 4.5.7
  creationTimestamp: "2020-08-20T09:50:48Z"
  generation: 2
  labels:
    samples.operator.openshift.io/managed: "true"
  managedFields:
  - apiVersion: image.openshift.io/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:openshift.io/display-name: {}
          f:samples.operator.openshift.io/version: {}
        f:labels:
          .: {}
          f:samples.operator.openshift.io/managed: {}
      f:spec:
        f:tags:
          .: {}
          k:{"name":"10"}:
            .: {}
            f:annotations:
              .: {}
              f:description: {}
              f:iconClass: {}
              f:openshift.io/display-name: {}
              f:openshift.io/provider-display-name: {}
              f:sampleRepo: {}
              f:tags: {}
              f:version: {}
            f:from:
              .: {}
              f:kind: {}
              f:name: {}
            f:generation: {}
            f:importPolicy: {}
            f:name: {}
            f:referencePolicy:
              .: {}
              f:type: {}
          k:{"name":"12"}:
            .: {}
            f:annotations:
              .: {}
              f:description: {}
              f:iconClass: {}
              f:openshift.io/display-name: {}
              f:openshift.io/provider-display-name: {}
              f:sampleRepo: {}
              f:tags: {}
              f:version: {}
            f:from:
              .: {}
              f:kind: {}
              f:name: {}
            f:generation: {}
            f:importPolicy: {}
            f:name: {}
            f:referencePolicy:
              .: {}
              f:type: {}
          k:{"name":"latest"}:
            .: {}
            f:annotations:
              .: {}
              f:description: {}
              f:iconClass: {}
              f:openshift.io/display-name: {}
              f:openshift.io/provider-display-name: {}
              f:sampleRepo: {}
              f:supports: {}
              f:tags: {}
            f:from:
              .: {}
              f:kind: {}
              f:name: {}
            f:generation: {}
            f:importPolicy: {}
            f:name: {}
            f:referencePolicy:
              .: {}
              f:type: {}
    manager: cluster-samples-operator
    operation: Update
    time: "2020-08-20T09:50:48Z"
  name: nodejs
  namespace: openshift
  resourceVersion: "18432"
  selfLink: /apis/image.openshift.io/v1/namespaces/openshift/imagestreams/nodejs
  uid: 526c38fd-dc56-40c3-b1dd-a9146d9b7ce5
spec:
  lookupPolicy:
    local: false
  tags:
  - annotations:
      description: Build and run Node.js 10 applications on RHEL 7. For more information
        about using this builder image, including OpenShift considerations, see https://github.com/sclorg/s2i-nodejs-container/blob/master/10/README.md.
      iconClass: icon-nodejs
      openshift.io/display-name: Node.js 10
      openshift.io/provider-display-name: Red Hat, Inc.
      sampleRepo: https://github.com/sclorg/nodejs-ex.git
      tags: builder,nodejs
      version: "10"
    from:
      kind: DockerImage
      name: registry.redhat.io/rhscl/nodejs-10-rhel7:latest
    generation: 2
    importPolicy: {}
    name: "10"
    referencePolicy:
      type: Local
  - annotations:
      description: Build and run Node.js 12 applications on RHEL 7. For more information
        about using this builder image, including OpenShift considerations, see https://github.com/sclorg/s2i-nodejs-container/blob/master/12/README.md.
      iconClass: icon-nodejs
      openshift.io/display-name: Node.js 12
      openshift.io/provider-display-name: Red Hat, Inc.
      sampleRepo: https://github.com/sclorg/nodejs-ex.git
      tags: builder,nodejs
      version: "12"
    from:
      kind: DockerImage
      name: registry.redhat.io/rhscl/nodejs-12-rhel7:latest
    generation: 2
    importPolicy: {}
    name: "12"
    referencePolicy:
      type: Local
  - annotations:
      description: |-
        Build and run Node.js applications on RHEL 7. For more information about using this builder image, including OpenShift considerations, see https://github.com/sclorg/s2i-nodejs-container/blob/master/12/README.md.

        WARNING: By selecting this tag, your application will automatically update to use the latest version of Node.js available on OpenShift, including major version updates.
      iconClass: icon-nodejs
      openshift.io/display-name: Node.js (Latest)
      openshift.io/provider-display-name: Red Hat, Inc.
      sampleRepo: https://github.com/sclorg/nodejs-ex.git
      supports: nodejs
      tags: builder,nodejs
    from:
      kind: ImageStreamTag
      name: "12"
    generation: 1
    importPolicy: {}
    name: latest
    referencePolicy:
      type: Local
status:
  dockerImageRepository: image-registry.openshift-image-registry.svc:5000/openshift/nodejs
  publicDockerImageRepository: default-route-openshift-image-registry.apps-crc.testing/openshift/nodejs
  tags:
  - items:
    - created: "2020-08-20T09:50:58Z"
      dockerImageReference: registry.redhat.io/rhscl/nodejs-10-rhel7@sha256:7a32c67dfa9f6f67a98f0c1eac5b68c088849abe4c9833d2fbdca7a10e01ed69
      generation: 2
      image: sha256:7a32c67dfa9f6f67a98f0c1eac5b68c088849abe4c9833d2fbdca7a10e01ed69
    tag: "10"
  - items:
    - created: "2020-08-20T09:50:58Z"
      dockerImageReference: registry.redhat.io/rhscl/nodejs-12-rhel7@sha256:2bfd71b41a1591a1ea19513cb92e5c1f43d235bbdf15b114b8fddd5521e99482
      generation: 2
      image: sha256:2bfd71b41a1591a1ea19513cb92e5c1f43d235bbdf15b114b8fddd5521e99482
    tag: "12"
  - items:
    - created: "2020-08-20T09:50:58Z"
      dockerImageReference: registry.redhat.io/rhscl/nodejs-12-rhel7@sha256:2bfd71b41a1591a1ea19513cb92e5c1f43d235bbdf15b114b8fddd5521e99482
      generation: 2
      image: sha256:2bfd71b41a1591a1ea19513cb92e5c1f43d235bbdf15b114b8fddd5521e99482
    tag: latest
```

**Comment by dharmit** [comment]
@prietyc123 

> I set the priority as high because its blocking 4.6 cluster to enable in periodic job against prow.

Are the periodic jobs still failing on this? Or did things get fixed with #4016 getting closed?

@dgolovin 

>  I am hitting the same problem for devfile nodejs components.
> Using crc v1.15.0 with OpenShift v4.5.7

Is this still a problem? Is it blocking VS Code plugin team? I'm asking this because I'm not seeing much activity on this issue and am inclined to lower its priority if it's not an annoyance to anyone. Plus, we're very low on bandwidth currently and need to focus on things that are absolute blockers/annoyance.

**Comment by prietyc123** [comment]
> Are the periodic jobs still failing on this? Or did things get fixed with #4016 getting closed?

No, It has been closed via https://github.com/openshift/odo/pull/4070 . I think we were facing the issue due to unsupported images. So closing this issue. @dgolovin please feel free to open this, if you still face the issue.
